### DISCUSSION TITLE
Different results running optimization several times

### DISCUSSION THREAD

**Discussion by vitalepaola** [BlueBrain/BluePyOpt#483] [General]
Hi, I'm encountering some troubles repeating the same optimization several times. Despite using the same cell_seed number, parameters, and number of generations, the best seeds are not always identical. I have also tried copying the same folder twice and running the optimization subsequently for just 5 generations. In both cases, the optimization stopped due to the "Max nGeneration criteria reached," but the best seed and evolution obtained were different. Is this reasonable?

**Comment by AurelienJaquier** [comment]
Hi @vitalepaola !
Indeed, for a same seed and number of generations, the output should be the same.
I was able to reproduce your issue, and to solve it easily by updating bluepyopt to its latest version (along with the dependencies that would no longer fulfill bluepyopt's requirements, i.e. Pebble and deap in your case), and by updating also numpy to its latest version.
Hope that helps!

**Comment by AurelienJaquier** [comment]
To be precise, I have used:
bluepyopt-1.14.7
deap-1.4.1
Pebble-5.0.6
numpy-1.26.3
